src/Util/DirectoryUtil.php

```diff
@@ -42,13 +42,35 @@ public static function stripAncestor(string $path, string $ancestor): string
      * Strips the trailing slash (directory separator) from a given path.
      *
      * @param string $path
-     *   Any path, absolute or relative, existing or not.
+     *   Any path, absolute or relative, existing or not. Empty paths and device
+     *   roots will be returned unchanged--but note that Windows UNC root path
+     *   detection is not robust. No validation is done to ensure that given
+     *   paths are valid.
      *
-     * @return string
+     * @see https://www.linux.com/training-tutorials/absolute-path-vs-relative-path-linuxunix/
+     * @see https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats
      */
     public static function stripTrailingSlash(string $path): string
     {
-        return rtrim($path, '/\\');
+        // Don't change an empty path.
+        if ($path === '') {
+            return '';
+        }
+
+        // Don't change a Windows drive-letter-or-UNC root path.
+        // @see https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch08s18.html
+        if (preg_match('/^([a-z]:\\\\?$)|(\\\\\\\\[a-z0-9_-]+\\\\[a-z0-9_-]+\\\\?$)/i', $path) === 1) {
```

Comment:
This is one hell of a regex. We should probably discuss it, and possibly document it more here, since it's terrifying to behold. You are clearly a braver man than I.

Reply:
Lol. You're right. Let's do discuss it. 👍